src/circle.rs

```diff
@@ -147,6 +150,132 @@ impl Iterator for CirclePathIter {
     }
 }
 
+/// A segment of a circle.
+///
+/// If `inner_radius > 0`, then the shape will be a doughnut segment.
+pub struct CircleSegment {
+    /// The center.
+    pub center: Point,
+    /// The outer radius.
+    pub outer_radius: f64,
+    /// The inner radius.
+    pub inner_radius: f64,
+    /// The angle to start drawing the segment (in radians).
+    pub start_angle: f64,
+    /// The arc length of the segment (in radians).
+    pub sweep_angle: f64,
+}
+
```

Comment:
It would be nice to have a `new` method here for constructing a segment. I could also imagine a convenience method on `Circle` that creates a `CircleSegment` given an inner radius, start angle, and sweep angle?

Reply:
I've added the methods, are they implemented how you would like?